src/New-InitialDotNetCoreProjects.ps1

```diff
@@ -0,0 +1,138 @@
+<#
+.SYNOPSIS
+Create an initial dotnet core project
+
+.DESCRIPTION
+Create an initial dotnet core project,  also populating the project guids from the solution file into the csproj files.
+
+.PARAMETER Prefix
+The prefix for the project name.  
+The script will create:
+
+A web project called $Prefix
+A unit test project called $Prefix.UnitTests
+A solution file called $Prefix.sln
+
+.PARAMETER ProjectType
+The type of project to create. Valid options are currently:
+* mvc, web, webapp
+* webapi
+* console
+* classlib
+* function
+
+.EXAMPLE
+New-InitialDotNetCoreProjects.ps1 --Prefix SomeProject --ProjectType web
+
+#>
+
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory=$true)]
+    [string] $Prefix,
+    [Parameter(Mandatory=$true)]
+    [ValidateSet("webapi","web","mvc","webapp", "console", "classlib", "function")]
+    [string] $ProjectType
+)
+
+function New-BasicProject {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $Name,
+        [Parameter(Mandatory=$true)]
+        [ValidateSet("webapi","web","mvc","webapp", "console", "classlib")]
+        [string] $ProjectType
+    )
+
+    $testProject = $Name + ".UnitTests"
+
+    & "dotnet" "new" $ProjectType "--name" $Name "--framework" netcoreapp3.1
+    & "dotnet" "new" "nunit" "--name" $testProject "--framework" netcoreapp3.1
+
+    & "dotnet" "new" "sln" "--name" $Name
+    & "dotnet" "sln" "add" "$($Name)/$($Name).csproj"
+    & "dotnet" "sln" "add" "$($testProject)/$($testProject).csproj"
+}
+
+function New-FunctionProject {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $Name
+    )
+
+    $testProject = $Name + ".UnitTests"
+
+    # Ensure the azure function dotnet cli template is installed
+    & "dotnet" "new" "-i" "Microsoft.Azure.WebJobs.ProjectTemplates::3.1.1624" | Out-Null
+
+    New-Item -Path $Name -ItemType Directory
+    Push-Location
+    Set-Location -Path $Name
+
+    & "dotnet" "new" "function" "--name" $Name "--AzureFunctionsVersion" V3
+
+    Pop-Location
+
+    & "dotnet" "new" "nunit" "--name" $testProject "--framework" netcoreapp3.1
```

Comment:
```suggestion
    & "dotnet" "new" "nunit" "--name" $TestProject "--framework" netcoreapp3.1
```
Can the variables be PascalCase? https://github.com/SkillsFundingAgency/das-platform-automation#naming-conventions

Reply:
Updated to meet our teams naming standards.